### PR TITLE
Remove auction support routes II

### PR DESCRIPTION
**Pull request by dblandin**
Follow-up to https://github.com/artsy/force/pull/4921

We've recently moved registration and bidding functionality to [artsy/reaction](https://github.com/artsy/reaction). These routes and components are no longer used and should be cleaned up.

jira :lock:: [AUCT-493][]

[AUCT-493]: https://artsyproduct.atlassian.net/browse/AUCT-493

Todo:

- [x] Deploy to new review app
- [x] QA registration and bidding on mobile web
- [x] QA registration and bidding on desktop web

Registration QA should include:
- No cards on file
- Cards on file
- From artwork page
- From sale page

**Screen Recordings**

**desktop web / register / card on file**

Peek 2020-01-15 15-58

---

**desktop web / register + bid / card on file**

Peek 2020-01-15 16-01

---

**desktop web / register / no card on file**

Peek 2020-01-15 16-07

---

**desktop web / register + bid / no card on file**

Peek 2020-01-15 16-09

---

**mobile web / register / card on file**

Peek 2020-01-15 16-26

---

**mobile web / register / no card on file**

Peek 2020-01-15 16-27

---

**mobile web / register + bid / card on file**

Peek 2020-01-15 16-28

---

**mobile web / register + bid / no card on file**

Peek 2020-01-15 16-31

### CODE REVIEW

File: src/mobile/apps/feature/routes.coffee [hunk 1]
```diff
@@ -1,133 +1,9 @@
...
 module.exports.index = (req, res, next) ->
   new Feature(id: req.params.id).fetch
     cache: true
     success: (feature) ->
       res.locals.sd.FEATURE = feature.toJSON()
       res.render 'page', feature: feature
-    error: res.backboneError
-
-module.exports.auctionRegister = (req, res, next) ->
-  # If not logged in, send them to sign up to return to this handler
```
Comment (erikdstock) [inline src/mobile/apps/feature/routes.coffee]: did we figure out where the `feature` app was used?

File: src/desktop/apps/auction_support/routes.coffee [hunk 2]
```diff
@@ -35,37 +29,24 @@ registerOrRender = (sale, req, res, next) ->
...
-        # TODO: When the bid + register flow is fully moved to reaction these routes
-        # and templates will need cleanup and templates will need deletion.
-        res.render 'registration',
-          sale: sale
-          monthRange: DateHelpers.getMonthRange()
-          yearRange: DateHelpers.getYearRange()
 
 
-@auctionRegistration = (req, res, next) ->
+@modalAuctionRegistration = (req, res, next) ->
```
Comment (dblandin) [inline src/desktop/apps/auction_support/routes.coffee]: @erikdstock This was removed in the diff in the last PR which broke registration on the sale page when a user already has a credit card on file. This route is currently triggered via the modal on the sale page, so I've renamed this route handler and kept the surrounding code (thankfully not much) in.

File: src/mobile/apps/feature/routes.coffee (hunk 1 above)
Comment (dblandin) [reply src/mobile/apps/feature/routes.coffee]: I've found a few recent examples:

https://www.artsy.net/feature/discover-art-from-stuttgart
https://www.artsy.net/feature/laguna-art-museum-california-cool-benefit-auction-2020
https://www.artsy.net/feature/whitechapel-gallery-art-icon-benefit-auction-2020

From what I can tell, no features currently support registration or bidding directly on the feature page. Auctions have a corresponding `Feature` but the feature page `/feature/:feature_slug` now redirects you to the `auction/:auction_slug` route.

**Comment by dblandin** [comment]
> I realize this is still pending due to merge conflicts but tentatively approve

I addressed those merge conflicts. Looks like https://github.com/artsy/force/pull/4933 updated a few files I removed or updated.

**Comment by zephraph** [comment]
Doing the Lord’s work here 🙌

🦾

File: src/desktop/analytics/artwork.js [hunk 3]
```diff
@@ -522,9 +522,4 @@ import { data as sd } from "sharify"
     .on("artwork:contact-specialist", function() {
       analytics.track("Clicked 'Contact Artsy Specialist'")
     })
-    .on("artwork:confirm-registration", function() {
```
Comment (dleve123) [inline src/desktop/analytics/artwork.js]: Do we know if there's any corresponding instrumentation in Reaction now? Was the only call-sight to this hook from code that was deleted in this PR? Just wanna make sure we're not deleting an analytics event

Comment (dblandin) [reply src/desktop/analytics/artwork.js]: Yes, we have page tracking in reaction. I don't see any explicit examples of this event triggered in either Force or Reaction. Might have been fired dynamically using the route and model passed in. Not sure but either way should be safe to remove.

Comment (dleve123) [reply src/desktop/analytics/artwork.js]: Yeah, I guess it wasn't be used for a while anyways in Force, even if the code was there.

**Comment by dblandin** [comment]
Merging this in :tada:

**Pull request merged**